ui/src/shared/components/AppContext.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from "react";
+
+type ContextProps = {
+  isDetailScreen?: boolean;
+  setIsDetailScreen?: React.Dispatch<boolean>;
+};
+
+const AppContext = React.createContext<ContextProps>({});
+
+export const AppContextProvider: React.FC = function ({ children }) {
```

Comment:
On post details screen, we can use the url to determine if we are in details screens

e.g.

```ts
const match = useRouteMatch("/posts/:postId");
if (match == null) {
  // not detail screen
} else {
  const postId = match.params.postId;
  // detail screen
}
```

Reply:
@lxcid Thanks for the tip, completely forgot about that. I've remove AppContext and used useRouteMatch instead.

Reply:
np! glad it help! :D